### PR TITLE
Implement point set simplification algorithms

### LINKED ISSUE #7: Point cloud resampling
See [downsampling](https://doc.cgal.org/latest/Point_set_processing_3/index.html#title37) and [upsampling](https://doc.cgal.org/latest/Point_set_processing_3/index.html#title53) from CGAL.

### PR DESCRIPTION
**Pull request by Q-Minh**
Fixes #7 

## Changes:
- WLOP resampling
- hierarchy simplification
- random simplification
- downsample executable for libigl visualization of WLOP, hierarchy and random downsampling

### CODE REVIEW

File: include/pcp/octree/linked_octree_node.hpp [hunk 1]
```diff
@@ -39,9 +39,6 @@ struct octree_parameters_t
     aabb_type voxel_grid{};            ///< The octree's bounding box
 };
 
-template <class Element, class ParamsType>
```
Comment (smercierpoirier) [inline include/pcp/octree/linked_octree_node.hpp:42]: My only question/concern, why did you remove the template of the iterator in the node?

**Review by smercierpoirier — COMMENTED** [review]
Apart from my comment, your code, comments and test LGTM.

File: include/pcp/octree/linked_octree_node.hpp (hunk 1 above)
Comment (Q-Minh) [reply include/pcp/octree/linked_octree_node.hpp:42]: Because the declaration + definition of the linked_octree_iterator is already done in the "linked_octree_iterator.hpp" file that is included at the top of the file. This declaration is redundant.

**Review by LongTran1996 — APPROVED** [review]
